selenium_tests/conftest.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+import os
+from unittest.mock import patch
+from types import SimpleNamespace
+
+from django.db import connection
+from django.conf import settings
+from selenium.webdriver import (
+    DesiredCapabilities,
+    Remote,
+)
+
+from selenium_tests.util import (
+    Browser,
+    DEFAULT_PASSWORD,
+    DatabaseLoader,
+    kill_connection_sql
+)
+from selenium_tests.data_util import create_user
+from selenium_tests.page import LoginPage
+from courses.factories import (
+    ProgramFactory,
+    CourseRunFactory,
+)
+from dashboard.models import ProgramEnrollment
+from financialaid.models import (
+    Tier,
+    TierProgram,
+)
+from search.indexing_api import (
+    delete_index,
+    recreate_index,
+)
+from roles.roles import Staff
+from roles.models import Role
+
+
+@pytest.fixture(scope='session')
+def django_db_setup(django_db_setup, django_db_blocker):
+    with django_db_blocker.unblock():
+        with connection.cursor() as cur:
+            cur.execute('DROP TABLE IF EXISTS wagtailsearch_editorspick CASCADE;')
+            DatabaseLoader().create_backup(db_cursor=cur)
+
+
+@pytest.fixture(autouse=True)
+def _db_loader(request, django_db_blocker):
+    marker = request.keywords.get('django_db', None)
+    if marker:
+        database_loader = DatabaseLoader()
+        with django_db_blocker.unblock():
+            with connection.cursor() as cur:
+                cur.execute(kill_connection_sql)
+            connection.close()
+        database_loader.load_backup()
+
+
+@pytest.fixture(scope='session')
+def internal_api_patcher():
+    methods_to_patch = [
+        'ecommerce.views.enroll_user_on_success',
+        'mail.api.MailgunClient._mailgun_request'
+    ]
+    patcher_mocks = []
+    patchers = [patch(method_name, autospec=True) for method_name in methods_to_patch]
+    for patcher in patchers:
+        mock = patcher.start()
+        mock.name = patcher.attribute
+        patcher_mocks.append(mock)
+    yield SimpleNamespace(
+        patchers=patchers,
+        patcher_mocks=patcher_mocks
+    )
+    for patcher in patchers:
+        patcher.stop()
+
+
+@pytest.fixture(autouse=True)
+def patches(internal_api_patcher):
+    for mock in internal_api_patcher.patcher_mocks:
+        mock.reset_mock()
+    return internal_api_patcher
+
+
+@pytest.fixture(scope='session')
+def driver():
+    # Start a selenium server running chrome
+    capabilities = DesiredCapabilities.CHROME.copy()
+    capabilities['chromeOptions'] = {
+        'binary': os.getenv('CHROME_BIN', '/usr/bin/google-chrome-stable'),
+        'args': ['--no-sandbox'],
+    }
+    driver = Remote(
+        os.getenv('SELENIUM_URL', 'http://chrome:5555/wd/hub'),
+        capabilities,
+    )
+    driver.implicitly_wait(10)
+    return driver
+
+
+@pytest.fixture(scope='session')
+def browser(driver, live_server):
+    return Browser(driver, live_server)
+
+
+@pytest.fixture()
+def es_index():
+    # TODO: need to recreate/delete for each test case?
```

Comment:
Yes, for the selenium tests it's a small part of the overhead, there's not much data to index at this point, and it helps ensure that tests don't communicate data across test boundaries